### PR TITLE
feat(Indavideó): add presence

### PR DESCRIPTION
**Pull request by techyyyt**
## Description 

Added presence for indavideo.hu.
Supported routes: /, /search, /video, /myprofile/basicdata, /myprofile/customize, /myprofile/myvideos, /myprofile/mylooprs, /myprofile/myplaylists, /myprofile/mywatchlater, /myprofile/mysubscribes, /myprofile/stripfactory
Supported subdomains: auto.indavideo.hu, film.indavideo.hu, csimota.indavideo.hu, palyazat.indavideo.hu, upload.indavideo.hu

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

</details>

### CODE REVIEW

File: websites/I/Indavideó/presence.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+				break;
+			}
+			case "upload.indavideo.hu": {
+				presenceData.details = "Uploading a video...";
+				break;
+			}
+		}
+	} else presenceData.details = "Browsing...";
+	presence.setActivity(presenceData);
+});
```
Comment (DarkVillager) [inline websites/I/Indavideó/presence.ts:106]:
```suggestion
presence.on("UpdateData", async () => {
	const presenceData: PresenceData = {
			largeImageKey: "https://i.imgur.com/WXPi1xl.png",
			startTimestamp: browsingTimestamp,
		},
		strings = await presence.getStrings({
			play: "general.playing",
			pause: "general.paused",
		}),
		{hostname, pathname} = document.location;
	if (
		pathname === "/" &&
		hostname === "indavideo.hu"
	)
		presenceData.details = "On the homepage";
	else if (
		pathname.includes("/search/text/") ||
		pathname.includes("/search/detailed/")
	) {
		searchBar = (<HTMLInputElement>document.querySelector("#ddl_search-what"))
			.value;
		presenceData.details = `Searching for "${searchBar}"...`;
	} else if (pathname.includes("/video/")) {
		delete presenceData.startTimestamp;
		(videoTitle = document.querySelector(".vid_header_title").textContent),
			(uploader = document.querySelector(".vid_header_username").textContent);
		presenceData.details = `Watching "${videoTitle.trim()}"`;
		presenceData.state = `Uploaded by ${uploader.trim()}`;
		presenceData.smallImageKey = paused ? Assets.Pause : Assets.Play;
		presenceData.smallImageText = paused ? strings.pause : strings.play;
		if (!paused) {
			const [startTimestamp, endTimestamp] = presence.getTimestamps(
				Math.floor(currentTime),
				Math.floor(duration)
			);
			presenceData.startTimestamp = startTimestamp;
			presenceData.endTimestamp = endTimestamp;
		}
		presenceData.buttons = [
			{
				label: "Watch video",
				url: toString(),
			},
		];
	} else if (pathname.includes("/profile/")) {
		uploader = document.querySelector("#user").textContent;
		presenceData.details = `Browsing user profile "${uploader.trim()}"...`;
	} else if (pathname.includes("/myprofile/basicdata"))
		presenceData.details = "Managing their account details...";
	else if (pathname.includes("/myprofile/customize"))
		presenceData.details = "Customizing their account...";
	else if (pathname.includes("/myprofile/myvideos"))
		presenceData.details = "Looking at their videos...";
	else if (pathname.includes("/myprofile/mylooprs"))
		presenceData.details = "Looking at their loops...";
	else if (pathname.includes("/myprofile/myplaylists"))
		presenceData.details = "Looking at their playlists...";
	else if (pathname.includes("/myprofile/mywatchlater"))
		presenceData.details = "Looking at their Watch Later list...";
	else if (pathname.includes("/myprofile/mysubscribes"))
		presenceData.details = "Looking at their subscriptions...";
	else if (pathname.includes("/myprofile/stripfactory"))
		presenceData.details = "Working on a strip embed...";
	else if (hostname !== "indavideo.hu") {
		switch (hostname) {
			case "auto.indavideo.hu": {
				presenceData.details = "Browsing the Autó category...";
				break;
			}
			case "film.indavideo.hu": {
				presenceData.details = "Browsing the Film category...";
				break;
			}
			case "csimota.indavideo.hu": {
				presenceData.details = "Browsing the Csimota category...";
				break;
			}
			case "palyazat.indavideo.hu": {
				presenceData.details = "Browsing the Pályázat category...";
				break;
			}
			case "upload.indavideo.hu": {
				presenceData.details = "Uploading a video...";
				break;
			}
		}
	} else presenceData.details = "Browsing...";
	presence.setActivity(presenceData);
});
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/I/Indavideó/presence.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+		pathname.includes("/search/text/") ||
+		pathname.includes("/search/detailed/")
+	) {
+		searchBar = (<HTMLInputElement>document.querySelector("#ddl_search-what"))
+			.value;
+		presenceData.details = `Searching for "${searchBar}"...`;
+	} else if (pathname.includes("/video/")) {
+		delete presenceData.startTimestamp;
+		(videoTitle = document.querySelector(".vid_header_title").textContent),
+			(uploader = document.querySelector(".vid_header_username").textContent);
```
Comment (DarkVillager) [inline websites/I/Indavideó/presence.ts:41]: Don't use `,` when you (re)assign let variables.
```suggestion
		videoTitle = document.querySelector(".vid_header_title").textContent;
			uploader = document.querySelector(".vid_header_username").textContent;
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/I/Indavideó/presence.ts (hunk 2 above)
Comment (techyyyt) [reply websites/I/Indavideó/presence.ts]: Fixed.

File: websites/I/Indavideó/metadata.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+	"thumbnail": "https://i.imgur.com/TvtJR3d.png",
+	"color": "#e74c3c",
+	"category": "videos",
+	"tags": [
+		"indavideo",
+		"hungarian",
+		"video"
+	],
+	"iframe": true,
+	"iFrameRegExp": ".*"
```
Comment (DarkVillager) [inline websites/I/Indavideó/metadata.json:23]: It's only one site so lets change the regex
```suggestion
	"iFrameRegExp": ".*[.]indavideo[.]hu"
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

**Review by theusaf — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**